src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int position = hashCode(key);
+        Node<K, V> currentNode = table[position];
+        if (currentNode == null) {
+            table[position] = new Node(key, value, null);
+            size++;
+            return;
+        } else {
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        table[position] = new Node(key, value, table[position]);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int position = hashCode(key);
+        Node<K, V> nodeByKey = table[position];
+        while (nodeByKey != null) {
+            if (Objects.equals(nodeByKey.key, key)) {
+                return nodeByKey.value;
+            }
+            nodeByKey = nodeByKey.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hashCode(Object key) {
```

Comment:
It's better to rename the method name. This method searches for the bucket number, not the hashcode itself

Reply:
I'd also recommend renaming this method to 'getIndexFromHash' or simply 'getIndex'. Also, replace 'Object' type with 'K'